diff --git a/src/Adapter/Hosting/HostingInformation.php b/src/Adapter/Hosting/HostingInformation.php
index f261abe40e2c9..84a261bfa3a98 100644
--- a/src/Adapter/Hosting/HostingInformation.php
+++ b/src/Adapter/Hosting/HostingInformation.php
@@ -89,4 +89,12 @@ public function isApacheInstawebModule()
     {
         return Tools::apacheModExists('mod_instaweb');
     }
+
+    /**
+     * @return string
+     */
+    public function getHostname(): string
+    {
+        return gethostname();
+    }
 }
diff --git a/src/Adapter/System/SystemInformation.php b/src/Adapter/System/SystemInformation.php
index fc567ed4c53d7..9f18199072d47 100644
--- a/src/Adapter/System/SystemInformation.php
+++ b/src/Adapter/System/SystemInformation.php
@@ -69,6 +69,7 @@ public function getSummary()
             'server' => $this->hostingInformation->getServerInformation(),
             'instaWebInstalled' => $this->hostingInformation->isApacheInstawebModule(),
             'uname' => $this->hostingInformation->getUname(),
+            'hostname' => $this->hostingInformation->getHostname(),
             'database' => $this->hostingInformation->getDatabaseInformation(),
             'overrides' => $this->shopInformation->getOverridesList(),
             'shop' => $this->shopInformation->getShopInformation(),
diff --git a/src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig b/src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig
index 9d2b5cba71828..01dad1cfdf58c 100644
--- a/src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig
+++ b/src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig
@@ -42,6 +42,11 @@
         <i class="material-icons">info_outline</i> {{ 'Server information'|trans }}
       </h3>
       <div class="card-body">
+        {% if system.hostname is not empty %}
+          <p class="mb-0">
+            <strong>{{ 'Server Hostname:'|trans }}</strong> {{ system.hostname }}
+          </p>
+        {% endif %}
         {% if system.uname is not empty %}
           <p class="mb-0">
             <strong>{{ 'Server information:'|trans }}</strong> {{ system.uname }}
